src/estimation/include/iDynTree/Estimation/AttitudeEstimator.h

```diff
@@ -0,0 +1,142 @@
+     */
+    class IAttitudeEstimator
+    {
+    public:
+        virtual ~IAttitudeEstimator();
+
+        /**
+         * @brief Update the filter with accelerometer and gyroscope measurements
+         *
+         * @param[in] linAccMeas left trivialized 3D vector of linear proper sensor acceleration measuements
+         * @param[in] gyroMeas left trivialized 3D vector of angular velocity measurements
+         *
+         * @note left trivialized angular velocity means the angular velocity of body frame B with respect to an inertial fram A, expressed in frame B
+         * @return true/false if successful/not
+         */
+        virtual bool updateFilterWithMeasurements(const iDynTree::LinearAccelerometerMeasurements& linAccMeas,
+                                          const iDynTree::GyroscopeMeasurements& gyroMeas) = 0;
+
+        /**
+         * @brief Update the filter with accelerometer, gyroscope and magnetometer measurements
+         *
+         * @param[in] linAccMeas left trivialized 3D vector of linear proper sensor acceleration measuements
+         * @param[in] gyroMeas left trivialized 3D vector of angular velocity measurements
+         * @param[in] magMeas left trivialized 3D vector of magnetometer measurements
+         *
+         * @note left trivialized angular velocity means the angular velocity of body frame B with respect to an inertial fram A, expressed in frame B
+         *
+         * @return true/false if successful/not
+         */
+        virtual bool updateFilterWithMeasurements(const iDynTree::LinearAccelerometerMeasurements& linAccMeas,
+                                                  const iDynTree::GyroscopeMeasurements& gyroMeas,
+                                                  const iDynTree::MagnetometerMeasurements& magMeas) = 0;
+
+        /**
+         * @brief Propagate the states and associated uncertainties through properly defined propagation functions
+         * The underlying implementation depends on the type of filter being implemented.
+         *
+         * @return true/false if successful/not
+         */
+        virtual bool propagateStates() = 0;
+
+        /**
+         * @brief Get orientation of the body with respect to inertial frame, in rotation matrix form
+         *
+         * @param[out] rot Rotation matrix
+         * @return true/false if successful/not
+         */
+        virtual bool getOrientationEstimateAsRotationMatrix(iDynTree::Rotation& rot) = 0;
+
+        /**
+         * @brief Get orientation of the body with respect to inertial frame, in quaternion form
+         *
+         * @note quaternion has the form (real, imaginary) and is normalized
+         *
+         * @param[out] q Quaternion
+         * @return true/false if successful/not
+         */
+        virtual bool getOrientationEstimateAsQuaternion(iDynTree::Quaternion& q) = 0;
+
+        /**
+         * @brief Get orientation of the body with respect to inertial frame, in Euler's RPY form
```

Comment:
Please add a reference to the GetRPY method of Rotation that provides a precise mathematical documentation of the RPY used here.